site/src/components/checkbox.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Checkbox
+description: The checkbox component in the Atlas Design System
+template: standard
+figmaEmbed: https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FuVA2amRR71yJZ0GS6RI6zL%2F%25F0%259F%258C%259E-Atlas-Design-Library%3Fnode-id%3D855%253A1014"
+---
+
+# Checkbox
+
+A selectable menu list from which an user can make one or more selections.
+
+<div class="table-wrapper margin-top-xs">
+	<table class="table">
+		<thead>
+			<tr>
+				<th>Type</th>
+				<th>Default</th>
+				<th>Disabled</th>
+			</tr>
+		</thead>
+		<tbody>
+			<tr>
+				<td>Unchecked</td>
+				<td>
+					<label class="checkbox">
+						<input type="checkbox" />
+						<span class="checkbox-check" role="presentation" aria-hidden="true"></span>
+						<span class="checkbox-text" aria-hidden="true">Default</span>
+					</label>
```

Comment:
Now that there's better support for [accent-color](https://developer.mozilla.org/en-US/docs/Web/CSS/accent-color) let's retire the special markup to enable comething more like this:
```html
<label class="checkbox"><input type="checkbox" /> Reindeer</label>
```

It would be worth considering whether we need a class at all or whether accent-color should be part of our base styles.

cc @wibjorn

Reply:
Without the custom checkbox, it would look like something on the left 
![image](https://user-images.githubusercontent.com/94572161/148844631-ed88a791-1a58-4d08-b40a-38e94324c889.png)

Here's a codepen to play with: https://codepen.io/homing1/pen/mdBzRLV

Some implications:
- The checkmark color will be defined by the accent color/browser. There will be no color difference from current behavior with all themes so this is ok.
- There will be a subtle background color change on hover checked, but looks ok.
- We would lose the grey checkmark on hover unchecked.

Should we loop in design? 
cc @Courtny

Reply:
Yes- please loop in design. I'm thinking the accent-color should be set globally.

```css
:root {
 accent-color: var(--theme-primary-base);
}

.checkbox2 {
 display: inline-flex;
 align-items: center;
 gap: 0.1875em;
}
```

With markup like this:

```html
<label class="checkbox2"><input type="checkbox"> Hello world</label>
```

Reply:
Moving discussion to here https://github.com/microsoft/atlas-design/discussions/152

Reply:
Seems to me the accent color approach has two major downsides:
- we still need to style the check, because the browser check is pretty ugly.
- we lose the smooth (albiet) subtle transition of the background color on state changes.

Reply:
It doesn't seem like there's a good overall solution to this. Design is against losing the color change on hover unchecked. Let's not pursue this approach until there's a way to make it work for our purposes.